homeassistant/components/portainer/binary_sensor.py

```diff
@@ -0,0 +1,161 @@
+
+        assert endpoint.containers
+        entities.extend(
+            [
+                PortainerContainerSensor(
+                    coordinator=coordinator,
+                    entity_description=entity_description,
+                    device_info=container,
+                    via_device=endpoint,
+                )
+                for container in endpoint.containers.values()
+                for entity_description in CONTAINER_SENSORS
+            ]
+        )
+
+    async_add_entities(entities)
+
+
+class PortainerEndpointSensor(PortainerEndpointEntity, BinarySensorEntity):
+    """Representation of a Portainer endpoint binary sensor entity."""
+
+    entity_description: PortainerBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: PortainerCoordinator,
+        entity_description: PortainerBinarySensorEntityDescription,
+        device_info: PortainerCoordinatorData,
+    ) -> None:
+        """Initialize Portainer endpoint binary sensor entity."""
+        self.entity_description = entity_description
+        super().__init__(device_info, coordinator)
+
+        self._attr_unique_id = f"{coordinator.config_entry.entry_id}_{device_info.id}_{entity_description.key}"
+
+    @property
+    def available(self) -> bool:
+        """Return if the device is available."""
+        return super().available and self.device_id in self.coordinator.data
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if the binary sensor is on."""
+
+        return (
+            self.entity_description.state_fn(device_info)
+            if (device_info := self.coordinator.data[self.device_id])
+            else None
+        )
+
+
+class PortainerContainerSensor(PortainerContainerEntity, BinarySensorEntity):
+    """Representation of a Portainer container sensor."""
+
+    entity_description: PortainerBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: PortainerCoordinator,
+        entity_description: PortainerBinarySensorEntityDescription,
+        device_info: DockerContainer,
+        via_device: PortainerCoordinatorData,
+    ) -> None:
+        """Initialize the Portainer container sensor."""
+        self.entity_description = entity_description
+        super().__init__(device_info, coordinator, via_device)
+
+        self._attr_unique_id = f"{coordinator.config_entry.entry_id}_{device_info.id}_{entity_description.key}"
+
+    @property
+    def available(self) -> bool:
+        """Return if the device is available."""
+        return super().available and self.endpoint_id in self.coordinator.data
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if the binary sensor is on."""
+
+        return (
+            self.entity_description.state_fn(device_info)
+            if (
+                device_info := self.coordinator.data[self.endpoint_id].containers.get(
+                    self.device_id
+                )
```

Comment:
to be honest, device sounds strange to me, as portainer works with containers, not with devices